build/linter/staticcheck/analyzer.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package staticcheck
+
+import (
+	"github.com/pingcap/tidb/build/linter/util"
+	"golang.org/x/tools/go/analysis"
+)
+
+var (
+	// Value to be added during stamping
+	name = "dummy value please replace using x_defs"
+
+	// Analyzer is an analyzer from staticcheck.
+	Analyzer *analysis.Analyzer
+)
+
+func init() {
+	Analyzer = FindAnalyzerByName(name)
```

Comment:
It seems the function `FindAnalyzerByName` can panic.

Reply:
It is a different way of building.  the ```name``` will be changed when compiled. the sub-lint of staticcheck is built from it. 

here is the building [script](https://github.com/pingcap/tidb/blob/6fc21e46c29aa730a0550b284d51ca0bfb2f623c/build/linter/staticcheck/BUILD.bazel)